### PR TITLE
feat(secret-service): provide HTTP 400 when scope is not found

### PR DESCRIPTION
**Pull request by thisthat**
## This PR

- When the scope is not found, the APIs return an HTTP 400 status code

Integration test run: https://github.com/keptn/keptn/actions/runs/2073164360

### CODE REVIEW

File: secret-service/pkg/backend/secretbackend_k8s_test.go [hunk 1]
```diff
@@ -422,6 +447,23 @@ func TestUpdateSecret_SecretNotFound(t *testing.T) {
 	assert.Equal(t, ErrSecretNotFound, err)
 }
 
+func TestUpdateSecret_ScopeNotFound(t *testing.T) {
```
Comment (odubajDT) [inline secret-service/pkg/backend/secretbackend_k8s_test.go:450]: Does it makes sense to to create a similar unit test also for creating a secret ?

Comment (odubajDT) [reply secret-service/pkg/backend/secretbackend_k8s_test.go:450]: ah sorry, It already exists, that's why I was missing it :)

**Review by odubajDT — APPROVED** [review]

File: secret-service/pkg/backend/secretbackend_k8s.go [hunk 2]
```diff
@@ -24,6 +25,30 @@ const SecretServiceName = "keptn-secret-service"
 var ErrSecretAlreadyExists = errors.New("secret already exists")
 var ErrSecretNotFound = errors.New("secret not found")
 var ErrTooBigKeySize = errors.New("name and key values must be no more than 253 characters")
+var ErrScopeNotFound = errors.New("")
```
Comment (warber) [inline secret-service/pkg/backend/secretbackend_k8s.go:28]:
```suggestion
var ErrScopeNotFound = errors.New("scope not found")
```

File: secret-service/pkg/backend/secretbackend_k8s.go [hunk 3]
```diff
@@ -24,6 +25,30 @@ const SecretServiceName = "keptn-secret-service"
...
+	}
+}
+
+func (se scopeErrorNotFound) Error() string {
+	return fmt.Sprintf("scope %s not available for creation of Secret %s", se.Scope, se.Secret)
+}
+
+func (se scopeErrorNotFound) Unwrap() error {
+	return se.Err
+}
```
Comment (warber) [inline secret-service/pkg/backend/secretbackend_k8s.go:51]: you can delete all of this

File: secret-service/pkg/backend/secretbackend_k8s.go [hunk 4]
```diff
@@ -46,7 +71,7 @@ func (k K8sSecretBackend) checkScopeDefined(secret model.Secret) (model.Scopes,
 	}
 	if _, ok := scopes.Scopes[secret.Scope]; !ok {
 		log.Errorf("Unable to find scope %s for secret %s", secret.Scope, secret.Name)
-		return model.Scopes{}, fmt.Errorf("scope %s not available for creation of Secret %s", secret.Scope, secret.Name)
+		return model.Scopes{}, NewScopeErrorNotFound(secret.Scope, secret.Name)
```
Comment (warber) [inline secret-service/pkg/backend/secretbackend_k8s.go:74]:
```suggestion
		return model.Scopes{}, fmt.Errorf("unable to check defined scope %s for secret %s: %w", secret.Scope, secret.Name, ErrScopeNotFound)
```

**Review by warber — COMMENTED** [review]
I think introducing a `scopeErrorNotFound` struct is not really necessary. The same can be accomplished with regular sentinel errors much more neatly, imo. see comments. :)

**Review by warber — APPROVED** [review]

**Pull request merged**